### PR TITLE
Support :json: dependencies

### PR DESCRIPTION
**Pull request by mmikitka**
PR for https://github.com/bazelbuild/rules_jvm_external/issues/351

1. Added json to SUPPORTED_PACKAGING_TYPES
2. Create genrule and alias for json deps
3. Do not include json deps in jvm_import deps list

### CODE REVIEW

**Comment by jin** [comment]
Could you add a integration test for a json dependency? See the WORKSPACE and `test/` directory for examples.

**Comment by mmikitka** [comment]
@jin Just added an integration test

File: WORKSPACE [hunk 1]
```diff
@@ -323,6 +323,29 @@ maven_install(
...
+        "io.quarkus:quarkus-maven-plugin:1.0.1.Final",
+        "io.quarkus:quarkus-bom-descriptor-json:1.0.1.Final",
+    ],
+    maven_install_json = "//tests/custom_maven_install:json_artifacts_testing_install.json",
+    repositories = [
+        "https://repo.maven.apache.org/maven2/",
+        "https://repo.spring.io/plugins-release/",
+    ],
+    fetch_sources = True,
+    use_unsafe_shared_cache = True,
```
Comment (jin) [inline WORKSPACE]: I don't think there's a need to test with fetch_sources and use_unsafe_shared_cache.

File: private/dependency_tree_parser.bzl [hunk 2]
```diff
@@ -91,6 +91,13 @@ def _generate_imports(repository_ctx, dep_tree, explicit_artifacts, neverlink_ar
         elif repository_ctx.attr.fetch_sources and ":sources:" in artifact["coord"]:
             # We already processed the sources above, so skip them here.
             pass
+        elif ":json:" in artifact["coord"]:
+            seen_imports[target_label] = True
+            versioned_target_alias_label = "%s_extension" % escape(artifact["coord"])
+            all_imports.append(
```
Comment (jin) [inline private/dependency_tree_parser.bzl]: hmm, seems weird to generate a target for a non-jar/aar/jvm package dependency. I can see this being confusing. Let's just skip generating targets for coords containing the `:json:` packaging?

I believe all you want to do is to fetch jar artifacts without the resolution breaking on artifacts with json dependencies, right?

Comment (artem-zinnatullin) [reply private/dependency_tree_parser.bzl]: Generic maven server supports arbitrary package types, I'd inflate other types as regular file targets
You never know what people download from maven haha

(ie OkBuck downloads Buck binary as `pex` package from maven)

Comment (jin) [reply private/dependency_tree_parser.bzl]: Right, sounds good, since the json artifact is marked as the primary artifact in `artifact["file"]`.

**Comment by mmikitka** [comment]
@jin I see your approval - is anything else required to merge this PR?

**Comment by jin** [comment]
Yes: https://github.com/bazelbuild/rules_jvm_external/pull/352#discussion_r362640437

**Comment by jin** [comment]
@mmikitka I think you pushed the wrong commit to this branch.

File: WORKSPACE (hunk 1 above)
Comment (mmikitka) [reply WORKSPACE]: I just removed `use_unsafe_shared_cache`

**Comment by mmikitka** [comment]
@jin My bad. I fixed up the branch and addressed your PR feedback

**Comment by jin** [comment]
Thanks. There's a conflict with master, please resolve that.

**Comment by mmikitka** [comment]
@jin I just resolved the merge conflict

**Comment by jin** [comment]
Thanks!

**Pull request merged**